src/integrations/prefect-dask/prefect_dask/task_runners.py

```diff
@@ -129,6 +132,18 @@ def result(
             _result = run_coro_as_sync(_result)
         return _result
 
+    def __del__(self):
+        if self._final_state or self._wrapped_future.done():
+            return
+        try:
+            local_logger = get_run_logger()
+        except Exception:
+            local_logger = logger
+        local_logger.warning(
+            "A future was garbage collected before it resolved."
+            " Please call `.wait()` or `.result()` on futures to ensure they resolve.",
+        )
```

Comment:
The way that we determine if a wrapped future is resolved is different for each future type. Also, `PrefectDistributedFuture` is a little tricky because sometimes it's ok if you don't wait on it (e.g. if `deferred` is `True`). I don't want to make a `__del__` method a required part of the interface until we can reliably check the resolution of all future types. I expect to revisit this again once `TaskRunWaiter` has some additional capabilities.

Reply:
Sounds good!